src/Chat.cs

```diff
@@ -17,7 +17,7 @@ public class Chat
 	/// <summary>
 	/// Gets or sets the messages of the chat history
 	/// </summary>
-	public List<Message> Messages { get; set; } = new();
+	public List<Message> Messages { get; set; } = [];
```

Comment:
We should consider changing this type. `List` is a bit too permissive in this public scope, especially when the only class adding messages is `Chat`. Switching to `IEnumerable`, `IAsyncEnumerable` (breaking), `IReadOnlyCollection`, or similar would make this class a bit _safer_. We can still utilize `List` internally, but I might also consider `LinkedList` or even `Queue` since we're always appending to the end and never directly modifying, accessing by index, or sorting anything.

Reply:
Yes, I am a fan of this approach, espacially `IReadOnlyCollection` to let uses of the class enumerate the items but cannot change its state. I had it this way but, @Plootie opened an issue #82 and Binozo submitted a pull request #62 to replace the message list.

It all went down to this: Developers might need to change the messages at any time. Inserting a system message or appending an assistant chunk for the LLM to continue. And they're developers after all, so they should know what they're doing (who am I to judge?) and might just need the control to do so.

Reply:
What if we return it to a `IReadOnlyCollection` but expose a constructor that allows users to provide their own source object? Similar to the constructor that was removed.

```
public class Program
{
	public static void Main()
	{
		
		var myList = new List<string>();
		var test = new InjectedListTest(myList);
		
		test.Add("First Item");		
		myList.Add("Added outside of test");		
		test.Add("Last Item");		
		
		Console.WriteLine(string.Join(", ", test.Collection));
	}
	
	private class InjectedListTest
	{
		private readonly ICollection<string> _base;
		public InjectedListTest(ICollection<string> myCollection)
		{
			_base = myCollection;
		}
		
		public void Add(string input) => _base.Add(input);
		public IEnumerable<string> Collection => _base;

	}
}
```

Reply:
The project I'm working on requires a lot of adding and removing messages (for example to fee the model "example" messages or to limit the history length). My needs may be unique and I understand if the desire is there to revert this back to a read-only.

Reply:
> The project I'm working on requires a lot of adding and removing messages (for example to fee the model "example" messages or to limit the history length). My needs may be unique and I understand if the desire is there to revert this back to a read-only.

I don't think your use case is out of the norm at all, but I also can't help but feel there is a graceful way we can build some structure around supporting that process without making the messages an easily-mutable list.

Reply:
I truly appreciate the thoughts you put into this detail as some might say, JD. As I said I was in that loop we well, not just once. But honestly I personally chose this way with exposing the list.
What is the purpose of the `Chat` class? It's a wrapper around the logic to build an interactive chat by using the api functions and keeping track of the messages sent back and forth. A helper, to make the  life of the developer a bit more convenient. So why should I bother to keep the developer from manipulating the list as they wish? It's not a list of business objects, bound data in a grid or anything like that. It's a collection of messages and after all I want any developer to use it has they want to.

Reply:
Thank you for your response and the background on the change. I understand and appreciate the desire to simplify usage for the end users, especially for a class that's largely a helper/convenience class for developers. I was confused with how the `Chat` class was being used and how it should be used by end users. I'm fearful of others having the same confusion. I had visualized `Chat` as a stateful object that could be used or extended for further business purposes later.